cmd/gapit/memory.go

```diff
@@ -0,0 +1,232 @@
+		if err != nil {
+			return log.Err(ctx, err, "Failed to load the capture")
+		}
+		verb.At = []uint64{uint64(boxedCapture.(*service.Capture).NumCommands) - 1}
+	}
+
+	boxedVal, err := client.Get(ctx, (&path.Metrics{
+		Command: capture.Command(verb.At[0], verb.At[1:]...),
+		Type:    path.Metrics_MEMORY,
+	}).Path())
+	if err != nil {
+		return log.Errf(ctx, err, "Failed to load metrics")
+	}
+
+	mem := boxedVal.(*service.Metrics).Metrics.(*service.Metrics_MemoryLayout).MemoryLayout
+
+	allocationFlags := []*service.Constant{}
+	if mem.AllocationFlagsIndex != -1 {
+		boxedConstants, err := client.Get(ctx, (&path.ConstantSet{
+			Api:   mem.Api,
+			Index: uint32(mem.AllocationFlagsIndex),
+		}).Path())
+		if err != nil {
+			return log.Errf(ctx, err, "Failed to load allocation flag names")
+		}
+		constants := boxedConstants.(*service.ConstantSet)
+		// If not a bitfield, we can't compare it against the flags
+		if constants.IsBitfield {
+			allocationFlags = constants.Constants
+		}
+	}
+
+	fmt.Printf("%v memory allocations\n", len(mem.Allocations))
+	sort.Slice(mem.Allocations, func(i, j int) bool {
+		return mem.Allocations[i].Id < mem.Allocations[j].Id
+	})
+	for _, alloc := range mem.Allocations {
+		fmt.Println("Name:", alloc.Name)
+		fmt.Println("\tDevice:      ", alloc.Device)
+		fmt.Println("\tMemory Type: ", alloc.MemoryType)
+		fmt.Println("\tSize:        ", alloc.Size)
+
+		if alloc.Flags != 0 && len(allocationFlags) != 0 {
+			fmt.Println("\tFlags:")
+			for _, f := range allocationFlags {
+				if (alloc.Flags & uint32(f.Value)) != 0 {
+					fmt.Printf("\t\t%v\n", f.Name)
+				}
+			}
+		}
+
+		if alloc.Mapping.Size != 0 {
+			fmt.Printf("\tMapped into host memory at 0x%x\n",
+				alloc.Mapping.HostAddress)
+			fmt.Println("\t\tOffset:", alloc.Mapping.Offset)
+			fmt.Println("\t\tSize:  ", alloc.Mapping.Size)
+		}
+
+		bindings := bindingSlice(alloc.Bindings)
+		sort.Slice(bindings, bindings.BindingLess)
+		fmt.Printf("\t%v Bindings:\n", len(bindings))
+		for _, binding := range bindings {
+			var typ string
+			switch binding.Type {
+			case service.MemoryBinding_BUFFER:
+				typ = "Buffer"
+			case service.MemoryBinding_IMAGE:
+				typ = "Image"
+			}
+			fmt.Printf("\t%v: %v\n", typ, binding.Name)
+
+			fmt.Println("\t\tOffset: ", binding.Offset)
+			fmt.Println("\t\tSize:   ", binding.Size)
+		}
+
+		aliases := bindings.computeAliasing()
+		if len(aliases) == 0 {
+			fmt.Println("\tNo aliased regions")
+		} else {
+			fmt.Printf("\t%v aliased regions:\n", len(aliases))
+			for i, a := range aliases {
+				fmt.Printf("\t%v:\n", i)
+				fmt.Println("\t\tOffset: ", a.offset)
+				fmt.Println("\t\tSize:   ", a.size)
+				fmt.Println("\t\tShared by:")
+				for _, s := range a.sharers {
+					fmt.Printf("\t\t\t%v\n", s)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+type bindingSlice []*service.MemoryBinding
+
+func (bindings bindingSlice) BindingLess(i, j int) bool {
+	if bindings[i].Offset < bindings[j].Offset {
+		return true
```

Comment:
Could simply, up to you
if bindings[i].Offset != bindings[j].Offset {
       return bindings[i].Offset < bindings[j].Offset
}